### PR TITLE
Fix contract tests for all Dynatrace resources

### PR DESCRIPTION
**Pull request by tbouron**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Dynatrace-Environment-ServiceLevelObjective/src/handlers.ts [hunk 1]
```diff
@@ -13,28 +26,98 @@ type SlosPayload = {
...
+                return progressEventBuilder.build();
+            } catch (e) {
+                logger.log(`Error ${e}`);
+                this.processRequestException(e, request);
+            }
+        }
+
+        try {
+            const getResponse = await this.get(model, typeConfiguration);
+            const getServerTiming = this.getServerTiming(getResponse);
```
Comment (joshuadeanhall) [inline Dynatrace-Environment-ServiceLevelObjective/src/handlers.ts:81]: It looks like we make this same call on line 60. Could we just get the `serverTiming` variable once and re-use that?

File: Dynatrace-Environment-SyntheticLocation/src/handlers.ts [hunk 2]
```diff
@@ -75,9 +75,13 @@ class Resource extends AbstractDynatraceResource<ResourceModel, LocationPayload,
...
         const resourceModel = new ResourceModel({
             ...model,
-            ...from
+            ...Transformer.for(from)
+                .transformKeys(CaseTransformer.IDENTITY)
+                .forModelIngestion()
+                .transform()
         });
         delete resourceModel.type_;
+        // delete resourceModel.locationNodeOutageDelayInMinutes;
```
Comment (joshuadeanhall) [inline Dynatrace-Environment-SyntheticLocation/src/handlers.ts:84]: Any reason to keep this code?

**Review by joshuadeanhall — CHANGES_REQUESTED** [review]

File: Dynatrace-Environment-ServiceLevelObjective/src/handlers.ts (hunk 1 above)
Comment (tbouron) [reply Dynatrace-Environment-ServiceLevelObjective/src/handlers.ts:81]: @joshuadeanhall This seems to not be available on all API calls. I would prefer to have this into the `AbstractDynatraceResource` but looks like we can't right now.

Comment (tbouron) [reply Dynatrace-Environment-ServiceLevelObjective/src/handlers.ts:81]: Well actually, I've coded this defencively so I can add `getServerTiming` method to `AbstractDynatraceResource`. Let me do that

Comment (tbouron) [reply Dynatrace-Environment-ServiceLevelObjective/src/handlers.ts:81]: @joshuadeanhall Done

**Review by joshuadeanhall — APPROVED** [review]

**Pull request merged**